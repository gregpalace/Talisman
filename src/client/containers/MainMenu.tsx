import React, { useState } from 'react';

import Flame from '../components/Flame';

import '../../stylesheets/mainMenu.scss';

const MainMenu = () => {
  const [isHovering, setHovering] = useState(false);

  const handleMouseHover = () => {
    return setHovering(!isHovering);
  };

  return (
    <div className='main-menu'>
      <header className='heading'>
        <img id='game-logo' src='../../assets/logo/logo.png' alt='Talisman' />
        <h1>Talisman</h1>
      </header>
      <br />
      <main className='menu-options'>
        <ul>
          <li>
            <a
              href='/game'
              onMouseEnter={handleMouseHover}
              onMouseLeave={handleMouseHover}
            >
              {' '}
              Start Game{' '}
              {isHovering && (
                <span>
                  <Flame />
                </span>
              )}
            </a>
          </li>
          <li>
            <a href='/game'>Continue Game</a>
          </li>
        </ul>
      </main>
    </div>
  );
};

export default MainMenu;
