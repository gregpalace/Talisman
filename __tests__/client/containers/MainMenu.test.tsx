import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import MainMenu from '../../../src/client/containers/MainMenu';

describe('React unit tests', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(<MainMenu />);
  });

  describe('<MainMenu />', () => {
    it('should render correctly', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should display the game logo in the header', () => {
      const header = wrapper.find('header');
      expect(header).toBeDefined();
      expect(header.find('img').prop('src')).toEqual(
        '../../assets/logo/logo.png'
      );
      expect(header.find('h1').text()).toEqual('Talisman');
    });

    it('should render a list with menu options', () => {
      const main = wrapper.find('main');
      expect(main).toBeDefined();
      expect(main.find('ul').length).toEqual(1);
    });

    it('should render both a start and continue game option', () => {
      const list = wrapper.find('ul');
      list.forEach((listItem: any, index: number) => {
        if (index === 0) {
          expect(listItem[index].prop('href')).toEqual('/game');
          expect(listItem[index].text()).toEqual(' Start Game ');
        }
      });
    });
  });
});
