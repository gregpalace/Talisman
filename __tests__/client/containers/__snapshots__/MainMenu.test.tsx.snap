// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React unit tests <MainMenu /> should render correctly 1`] = `
<div
  className="main-menu"
>
  <header
    className="heading"
  >
    <img
      alt="Talisman"
      id="game-logo"
      src="../../assets/logo/logo.png"
    />
    <h1>
      Talisman
    </h1>
  </header>
  <br />
  <main
    className="menu-options"
  >
    <ul>
      <li>
        <a
          href="/game"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
           
          Start Game
           
        </a>
      </li>
      <li>
        <a
          href="/game"
        >
          Continue Game
        </a>
      </li>
    </ul>
  </main>
</div>
`;
